subsys/net/l2/ethernet/ethernet.c

```diff
@@ -145,6 +145,22 @@ static inline bool eth_is_vlan_tag_stripped(struct net_if *iface)
 	return (api->get_capabilities(dev) & ETHERNET_HW_VLAN_TAG_STRIP);
 }
 
+/* Drop packet if it has broadcast destination MAC address but the IP
+ * address is not multicast or broadcast address. See RFC 1122 ch 3.3.6
+ */
+static inline enum net_verdict ipv4_check_bcast_addr(struct net_pkt *pkt)
+{
+	if (net_eth_is_addr_broadcast((struct net_eth_addr *)
+				      &NET_ETH_HDR(pkt)->dst.addr) &&
```

Comment:
you cannot do that since you pulled the ethernet header from the pkt (see line 259). Get the ethernet header as a paremeter, ethernet_recv() can provide it since it keeps a pointer to it in 'hdr' variable.

Reply:
Yep, that is right. Weirdly it worked just fine when I was testing it. Anyway, I will send a new version shortly.